internal/hashing/argon2id.go

```diff
@@ -0,0 +1,162 @@
+package hashing
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"golang.org/x/crypto/argon2"
+)
+
+const ARGON2_VERSION = 19
+
+type Argon2IDParameters struct {
+	SaltLength uint32
+	Time       uint32
+	Memory     uint32
+	Threads    uint8
+	KeyLength  uint32
```

Comment:
Should we specify the units next to the parameters? Ex:

```go

type Argon2IDParameters struct {
	SaltLength uint32 // bytes
}
```

Reply:
Yes, I think that would be useful. I will add that.

Reply:
I added the units.